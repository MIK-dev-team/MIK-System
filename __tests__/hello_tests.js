import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import ReservationTable from '../app/javascript/components/reservations_list/reservation_table';

describe('Welcome', () => {
    it('to the world of testing react', () => {
        expect(true).toBe(true);
    });

    it('to snapshot testing', () => {
        const component = renderer.create(
            <ReservationTable reservations={[]} />
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('to using enzyme', () => {
        let table = shallow(<ReservationTable reservations={[]} />);
        expect(table.find('h1').text()).toEqual('Varaukset');
    });

    it('to finding arrays', () => {
        let table = shallow(<ReservationTable reservations={[]} />);
        expect(table.find('Table').length).toEqual(1);
        expect(table.find('thead').length).toEqual(1);
        expect(table.find('th').length).toEqual(5);
    })
});