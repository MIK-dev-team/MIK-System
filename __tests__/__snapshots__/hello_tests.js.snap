// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome to snapshot testing 1`] = `
<div>
  <h1>
    Varaukset
  </h1>
  <div
    className="table-responsive"
  >
    <table
      className="table table-striped table-hover"
    >
      <thead>
        <tr>
          <th>
            #
          </th>
          <th>
            Alkamisaika
          </th>
          <th>
            Loppumisaika
          </th>
          <th>
            Kone
          </th>
          <th>
            Lennon tyyppi
          </th>
        </tr>
      </thead>
      <tbody>
        <tr />
      </tbody>
    </table>
  </div>
</div>
`;
